var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -1003,7 +1003,7 @@ def configure_args(self):
 
         # Work around incompatibility with new apple-clang linker
         # https://github.com/open-mpi/ompi/issues/12427
-        if spec.satisfies("@5: %apple-clang@15:"):
+        if spec.satisfies("@:4.1.6,5.0.0:5.0.3 %apple-clang@15:"):
             config_args.append("--with-wrapper-fcflags=-Wl,-ld_classic")
```

Comment:
This was a compiler flag that was removed, not a duplicate RPATH issue. So the fix is still needed in versions of OpenMPI before this was fixed.